### ISSUE TITLE
Unable to start synergy server with "Cannot create a file when that file already exists" message

### ISSUE THREAD

**Issue by nbolton** [symless/synergy-core#3752]
**Imported issue:**
- Author: ??? ???
- Date: 2013-11-08 06:59:47
- Legacy ID: 3822

**Steps to reproduce:**
1. Install Synergy, click "start"

Expected: Synergy should be start up

```
Actual: Below error observed & Synergy service cannot boot up
INFO: service command updated
ERROR: failed to launch, error: Cannot create a file when that file already exists
```

**Versions and operating systems:**

1.4.15 beta, Windows 7 Enterprise x64
1.5.0, Windows 7 Enterprise x64

**Temporary workarounds:**

Start synergys manually without --ipc option,
ex)

> "C:\Program Files\Synergy\synergys.exe" -f --no-tray --name NotePC --stop-on-desk-switch --enable-drag-drop -c C:/ProgramData/Synergy/synergy.sgc --address :24801

**Similar bugs:**

http://synergy-project.org/spit/issues/details/3842/

**Additional comments:**

N/A

**Comment by Qbuer** [comment]
work for me

**Comment by xpac1985** [comment]
I suddenly experience the same problem, I'm not perfectly sure but I think it started after I clicked the Start button in the GUI two or three times without waiting for it to respond. I tried rebooting.
Synergy is 1.8.8 stable and registered to Synergy Pro.

Here is what the DEBUG setting produces. I removed the serial for obvious reasons.

`[2017-04-18T10:40:12] ERROR: ipc connection error, connection refused
[2017-04-18T10:40:13] DEBUG: adopting new buffer
[2017-04-18T10:40:13] INFO: using last known command: "C:/Program Files/Synergy/synergys.exe" -f --no-tray --debug DEBUG --name trono --ipc --enable-drag-drop --enable-crypto --profile-dir C:\Users\xpac\AppData\Local -c C:/Users/xpac/AppData/Local/Temp/qt_temp.Ya4784 --address :24800 --serial-key ***XYZ***
[2017-04-18T10:40:13] INFO: service command updated
[2017-04-18T10:40:13] DEBUG: event queue is ready
[2017-04-18T10:40:13] DEBUG: add pending events to buffer
[2017-04-18T10:40:13] DEBUG: accepted ipc client connection
[2017-04-18T10:40:13] DEBUG: start ipc handle data
[2017-04-18T10:40:13] DEBUG: ipc read: IHEL
[2017-04-18T10:40:13] DEBUG: finished ipc handle data
[2017-04-18T10:40:13] DEBUG: found sas.dll
[2017-04-18T10:40:13] DEBUG: ipc hello, type=gui
[2017-04-18T10:40:13] INFO: watchdog status: error
[2017-04-18T10:40:13] DEBUG: getting elevated token, elevation required
[2017-04-18T10:40:13] DEBUG: processes in session 1: csrss.exe, winlogon.exe, fontdrvhost.exe, dwm.exe, NVDisplay.Container.exe, sihost.exe, svchost.exe, nvcontainer.exe, svchost.exe, taskhostw.exe, MoveToDesktop.exe, TeamViewer.exe, explorer.exe, MoveToDesktop.x64.exe, ShellExperienceHost.exe, SearchUI.exe, RuntimeBroker.exe, tv_w32.exe, tv_x64.exe, nvtray.exe, MSASCuiL.exe, RtkNGUI64.exe, AsusAudioCenter.exe, VeraCrypt.exe, SpotifyWebHelper.exe, DTAgent.exe, seafile-applet.exe, ccnet.exe, conhost.exe, BoxcryptorClassic.exe, seaf-daemon.exe, conhost.exe, Listary.exe, ListaryHelper64.exe, ListaryHookHelper32.exe, ListaryHookHelper64.exe, KeePass.exe, CCleaner64.exe, pdf24.exe, svchost.exe, NVIDIA Web Helper.exe, conhost.exe, ApplicationFrameHost.exe, WinStore.App.exe, SystemSettings.exe, thunderbird.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, dllhost.exe, chrome.exe, mmc.exe, chrome.exe, chrome.exe, backgroundTaskHost.exe, notepad.exe, synergy.exe, chrome.exe, backgroundTaskHost.exe, 
[2017-04-18T10:40:13] DEBUG: found winlogon.exe in session 1
[2017-04-18T10:40:13] DEBUG: got token 724, duplicating
[2017-04-18T10:40:13] DEBUG: duplicated, new token: 712
[2017-04-18T10:40:13] INFO: starting new process
[2017-04-18T10:40:13] DEBUG: launched syntool to check active desktop
[2017-04-18T10:40:13] INFO: activeDesktop:Default
[2017-04-18T10:40:14] DEBUG: getting elevated token, elevation required
[2017-04-18T10:40:14] DEBUG: processes in session 1: csrss.exe, winlogon.exe, fontdrvhost.exe, dwm.exe, NVDisplay.Container.exe, sihost.exe, svchost.exe, nvcontainer.exe, svchost.exe, taskhostw.exe, MoveToDesktop.exe, TeamViewer.exe, explorer.exe, MoveToDesktop.x64.exe, ShellExperienceHost.exe, SearchUI.exe, RuntimeBroker.exe, tv_w32.exe, tv_x64.exe, nvtray.exe, MSASCuiL.exe, RtkNGUI64.exe, AsusAudioCenter.exe, VeraCrypt.exe, SpotifyWebHelper.exe, DTAgent.exe, seafile-applet.exe, ccnet.exe, conhost.exe, BoxcryptorClassic.exe, seaf-daemon.exe, conhost.exe, Listary.exe, ListaryHelper64.exe, ListaryHookHelper32.exe, ListaryHookHelper64.exe, KeePass.exe, CCleaner64.exe, pdf24.exe, svchost.exe, NVIDIA Web Helper.exe, conhost.exe, ApplicationFrameHost.exe, WinStore.App.exe, SystemSettings.exe, thunderbird.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, chrome.exe, dllhost.exe, chrome.exe, mmc.exe, chrome.exe, chrome.exe, backgroundTaskHost.exe, notepad.exe, synergy.exe, chrome.exe, backgroundTaskHost.exe, 
[2017-04-18T10:40:14] DEBUG: found winlogon.exe in session 1
[2017-04-18T10:40:14] DEBUG: got token 796, duplicating
[2017-04-18T10:40:14] DEBUG: duplicated, new token: 712
[2017-04-18T10:40:14] INFO: starting new process
[2017-04-18T10:40:14] ERROR: could not launch
[2017-04-18T10:40:14] ERROR: exit code: 0
[2017-04-18T10:40:14] ERROR: failed to launch, error: Eine Datei kann nicht erstellt werden, wenn sie bereits vorhanden ist.
[2017-04-18T10:40:14] INFO: backing off, wait=2s, failures=1
`

**Comment by xpac1985** [comment]
What fixed it for me:
Copy the command line used by Synergy (at the very top of the log, in my case), remove the serial-key part.
Open a DOS box (cmd), paste the line. Watch client reconnect, see success in Synergy log window too.
Close cmd, client reconnects, everything is fine.
Another thing I tried, and which might have overlapped - activating Synergy again using the menu and my serial key. Cannot explain why that might have worked, but it did. ;)

**Comment by xpac1985** [comment]
About the re-registration:
Before the re-registration, the command line looked like shown above, but the Serial key was like:
123962592439(first parts of key)FFFFFFFFFFFFFFFFFF(about ~60.000 F's - yes, seriously)123973458973498534457234(remaining part of key).
After re-registration, the command line shows the proper serial key.

**Comment by NicolasGoeddel** [comment]
I was not able to re-register because the window freezes when I tried to click on File -> Activate.
So I cleared the corresponding registry value manually. Go to HKEY_CURRENT_USER/Software/Synergy/Synergy and make sure the entry "serialKey" is empty.

**Comment by xpac1985** [comment]
This is what my key window looks like, and the FFFFF-part tends to become longer every time.

![synergy_2017-04-28_16-01-24](https://cloud.githubusercontent.com/assets/6882771/25532518/ebd8a594-2c2d-11e7-95b3-33564ec85df8.png)

**Comment by NicolasGoeddel** [comment]
Is there a bug fix available? Because that bug is annoying. Every month or more frequent I have to reset the key in the registry because Synergy is not starting anymore.

**Comment by xpac1985** [comment]
I gave up on the Synergy team to fix any of the bugs I reported, and switched to Mouse without Borders. Feels like money wasted.

**Comment by NicolasGoeddel** [comment]
This is no option for me because I need it to control Linux and Windows PCs. On Linux this issue does not exist.

**Comment by nlyan** [comment]
This bug is fixed in the v1.9 release candidate builds, which are available on the forum under General Discussion
